lib/os/hash_map_oa_lp.c

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <zephyr/sys/hash_map.h>
+#include <zephyr/sys/hash_map_oa_lp.h>
+#include <zephyr/sys/util.h>
+
+enum bucket_state {
+	UNUSED,
+	USED,
+	TOMBSTONE,
+};
+
+struct oalp_entry {
+	uint64_t key;
+	uint64_t value;
+	enum bucket_state state;
+};
```

Comment:
This is probably the most simple way to store the keys, which is fine, but makes me want to try storing the keys and values separately, and encoding the bucket state in the key, and adding another benchmark comparision hehe

Reply:
Yep - I thought about that as well. It would be a good experiment, because padding the keys / values just seems wrong.